.github/workflows/main.yaml

```diff
@@ -349,3 +351,57 @@ jobs:
         run: |
           export DOCKER_BUILD_AND_SYSTEM_PRUNE="true"
           make image 
+  container_runtime_test:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        container-runtime: [ 'cri-o', 'docker' ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install and configure container runtime
+        run: |
+          if [ "${{ matrix.container-runtime }}" == "docker" ]; then
+            sudo apt-get update
+            sudo apt-get install \
+            apt-transport-https \
+            ca-certificates \
+            curl \
+            gnupg \
+            lsb-release
+            curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+            echo \
+            "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+             $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+            sudo apt-get update
+            sudo apt-get install docker-ce docker-ce-cli containerd.io  
+            git clone https://github.com/Mirantis/cri-dockerd.git -b ${{ env.CRIDOCKERD_VERSION }}
+            cd cri-dockerd
+            make cri-dockerd
+            sudo install -o root -g root -m 0755 cri-dockerd /usr/local/bin/cri-dockerd
+            sudo install packaging/systemd/* /etc/systemd/system
+            sudo sed -i -e 's,/usr/bin/cri-dockerd,/usr/local/bin/cri-dockerd,' /etc/systemd/system/cri-docker.service
+            sudo systemctl daemon-reload
+            sudo systemctl enable --now cri-docker.socket
+            sudo systemctl restart cri-docker
+            cd .. && sudo rm -rf cri-dockerd
+          elif [ "${{ matrix.container-runtime }}" == "cri-o" ]; then
+            sudo rm -rf cri-o.amd64.v${{ env.CRIO_VERSION }}.tar.gz && sudo rm -rf cri-o
+            sudo wget https://storage.googleapis.com/cri-o/artifacts/cri-o.amd64.v${{ env.CRIO_VERSION }}.tar.gz
+            sudo tar -zxvf cri-o.amd64.v${{ env.CRIO_VERSION }}.tar.gz
+            sudo sed -i 's/\/usr\/bin\/env sh/!\/bin\/bash/' cri-o/install   
+            sudo sed -i 's/ExecStart=.*/ExecStart=\/usr\/local\/bin\/crio --selinux=false \\/' cri-o/contrib/crio.service
+            cd cri-o
+            sudo /bin/bash ./install
+            sudo systemctl daemon-reload
+            sudo systemctl enable --now crio
+            sudo systemctl restart crio
+            cd .. && sudo rm -rf cri-o.amd64.v${{ env.CRIO_VERSION }}.tar.gz && sudo rm -rf cri-o
+          fi
+      - name: Run e2e
+        run: make e2e
+        env:
+          CONTAINER_RUNTIME: ${{ matrix.container-runtime }}
```

Comment:
after execute the job, please add a step to upload the logs

![image](https://github.com/kubeedge/kubeedge/assets/48788150/7b951e4f-a264-4703-afcb-0173607eaa7f)

Reply:
ok,I have made it.